### PR TITLE
Move closer refactoring indent fix

### LINKED ISSUE #3964: "Move closer to usage" refactor leaves a bit undone
It does not trigger a reparse and it does not execute the indenter on the changed method.

```vba
Private Sub DeleteExtraLogFiles(ByVal filePath As String, ByVal fileNamePattern As String, ByVal maxLogsToKeep As Long)

'move this vvv
  Dim oldestFile As String
  
  On Error GoTo ErrorHandler
  
  Dim fileCount As Long
  
  Dim currentFile As String
  currentFile = Dir(filePath & fileNamePattern)
  
  Dim oldestFileDate As Date
  oldestFileDate = #12/31/3999#
  
  Dim currentFileDate As Date
  
  Do While Len(currentFile) > 0
    fileCount = fileCount + 1
    currentFileDate = FileDateTime(filePath & currentFile)
    If fileCount < maxLogsToKeep Then
      If currentFileDate < oldestFileDate Then
        oldestFile = currentFile
        oldestFileDate = currentFileDate
      End If
    Else
...
```

Yields this when moving `Dim oldestFile As String`

```vba
Private Sub DeleteExtraLogFiles(ByVal filePath As String, ByVal fileNamePattern As String, ByVal maxLogsToKeep As Long)

  On Error GoTo ErrorHandler
  
  Dim fileCount As Long
  
  Dim currentFile As String
  currentFile = Dir(filePath & fileNamePattern)
  
  Dim oldestFileDate As Date
  oldestFileDate = #12/31/3999#
  
  Dim currentFileDate As Date
  
  Do While Len(currentFile) > 0
    fileCount = fileCount + 1
    currentFileDate = FileDateTime(filePath & currentFile)
    If fileCount < maxLogsToKeep Then
      If currentFileDate < oldestFileDate Then
'get this vvv
        Dim oldestFile As String
oldestFile = currentFile
'and that ^^^
        oldestFileDate = currentFileDate
      End If
    Else
...
```

The lack of reparse:
rd move closer

It also somewhat inconveniently leaves the cursor at the top of the module instead of somewhere in the method that's being edited.

### PR DESCRIPTION
**Pull request by tommy9**
Fixes #3964 

First line after insertion was given no indentation. This PR checks the indentation of the line prior to the insertion and adds padding so that the same indentation is preserved.

A reparse is also requested after the refactoring has completed.

### CODE REVIEW

**Comment by Vogel612** [comment]
For the Record: When fixing the original bug I tried to implement this on the parse tree, but gave it up. 
Directly accessing the codepane is discouraged, but I haven't found a clean way to do this ....

File: Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs [hunk 1]
```diff
@@ -168,9 +169,12 @@ private void InsertNewDeclaration()
             }
 
             var insertionIndex = (expression as ParserRuleContext).Start.TokenIndex;
+            var firstReferenceLine = _vbe.ActiveCodePane.CodeModule.GetLines((expression as ParserRuleContext).Start.Line, 1);
```
Comment (bclothier) [inline Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: `_vbe`., `ActiveCodePane`, and `CodeModule` each are a `SafeComWrapper`. Those must be wrapped in a `using` block for each reference to ensure proper release of the wrapper.

File: Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs [hunk 2]
```diff
@@ -168,9 +169,12 @@ private void InsertNewDeclaration()
...
+            var indentLength = firstReferenceLine.Length - firstReferenceLine.TrimStart().Length;
+            var padding = new string(' ', indentLength);
```
Comment (bclothier) [inline Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: I observe that none of unit tests account for the case where there is something that isn't "indentable" -- such as line numbers for example. Thus, what happens when we have code like `100:     Foo = bar`? The assumption here is that we only indent if there is only whitespace on the line and that might be actually fine but the unit tests should prove that assumption.

File: Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs [hunk 3]
```diff
@@ -168,9 +169,12 @@ private void InsertNewDeclaration()
...
 
             var rewriter = _state.GetRewriter(firstReference.QualifiedModuleName);
-            rewriter.InsertBefore(insertionIndex, newVariable);
+            rewriter.InsertBefore(insertionIndex, newVariable + padding);
```
Comment (bclothier) [inline Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: I wonder if given the fact that we create a `newVariable` which ends with a newline, it means that we can just look at the `expression`'s `Column` property and pad that many? That would avoid the COM access altogether.

File: Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs (hunk 2 above)
Comment (retailcoder) [reply Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: Very good point, but IMO not a showstopper - let's get the wrappers into `using` blocks, and if the line number stuff can't be fixed right away I'd still merge it.

File: Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs (hunk 1 above)
Comment (bclothier) [reply Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: PS I spoke a bit too hastily - `_vbe` doesn't need to be in the `using`; the other two do, though.

Comment (tommy9) [reply Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: Thanks, will update with a pair of ```using``` blocks

File: Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs (hunk 2 above)
Comment (tommy9) [reply Rubberduck.Core/Refactorings/MoveCloserToUsage/MoveCloserToUsageRefactoring.cs]: I've added a new test and it moves the line with the line number down without introducing any indent before the number so I think that's okay.

However, I tried a different test where the declaration has a line number. The declaration gets moved but leaves the line number behind. This is a problem when the line after the declaration has a line number too and suddenly we have a line with two line numbers! This is unrelated to the changes I made, so I will open a new issue.

**Comment by Hosch250** [comment]
Good job.  Feel free to join us in our war room: https://chat.stackexchange.com/rooms/14929/vba-rubberducking

**Pull request merged**